edit/configmap.vue

```diff
@@ -1,70 +1,99 @@
+import CruResource from '@/components/CruResource';
 import NameNsDescription from '@/components/form/NameNsDescription';
-import Footer from '@/components/form/Footer';
 import KeyValue from '@/components/form/KeyValue';
-import ResourceTabs from '@/components/form/ResourceTabs';
+import Labels from '@/components/form/Labels';
 import Tab from '@/components/Tabbed/Tab';
+import Tabbed from '@/components/Tabbed';
 
 export default {
   name: 'CruConfigMap',
 
   components: {
+    CruResource,
     NameNsDescription,
     KeyValue,
-    Footer,
-    ResourceTabs,
+    Labels,
     Tab,
+    Tabbed
   },
 
   mixins: [CreateEditView],
+
+  data() {
+    if (!this?.value?.spec?.type && !this.value?.spec) {
+      this.$set(this.value, 'spec', {});
+    }
+
+    return {};
+  },
 };
 </script>
 
 <template>
-  <form>
-    <NameNsDescription
-      :value="value"
-      :mode="mode"
-      name-label="Name"
-      :register-before-hook="registerBeforeHook"
-    />
-
-    <div class="spacer"></div>
-
-    <div v-if="!isView || Object.keys(value.data||{}).length" class="row">
-      <KeyValue
-        key="data"
-        v-model="value.data"
+  <CruResource
+    :done-route="doneRoute"
+    :mode="mode"
+    :resource="value"
+    :subtypes="[]"
+    :validation-passed="true"
+    :errors="errors"
+    @error="e=>errors = e"
+    @finish="save"
+    @cancel="done"
+  >
+    <template #define>
+      <NameNsDescription
+        v-if="!isView"
+        :value="value"
         :mode="mode"
-        :title="t('configmapPage.data.title')"
-        :protip="t('configmapPage.data.protip')"
-        :initial-empty-row="true"
+        name-label="Name"
+        :register-before-hook="registerBeforeHook"
       />
-    </div>
 
-    <div class="spacer"></div>
+      <div class="spacer"></div>
 
-    <ResourceTabs v-model="value" :mode="mode">
-      <template #before>
-        <Tab :label="t('configmapPage.tabs.binaryData.label')" name="binary-data">
+      <Tabbed :side-tabs="true">
+        <Tab name="data" :label="t('configmap.tabs.data.label')" :weight="1">
           <KeyValue
-            key="binaryData"
-            v-model="value.binaryData"
-            :protip="false"
+            key="data"
+            v-model="value.data"
             :mode="mode"
-            :add-allowed="false"
-            :read-accept="'*'"
-            :read-multiple="true"
-            :value-binary="true"
-            :value-base64="true"
-            :value-can-be-empty="true"
-            :initial-empty-row="false"
+            :protip="t('configmapPage.data.protip')"
+            :initial-empty-row="true"
           />
         </Tab>
-      </template>
-    </ResourceTabs>
-
-    <Footer :mode="mode" :errors="errors" @save="save" @done="done" />
-  </form>
+        <Tab name="binary-data" :label="t('configmap.tabs.binaryData.label')" :weight="2">
+          <div v-if="!isView || Object.keys(value.data||{}).length" class="row">
+            <KeyValue
```

Comment:
Editing non-ascii-like binary data should show you the length and not the raw value.

![image](https://user-images.githubusercontent.com/753917/90072705-c3760a00-dcac-11ea-9d5b-6c4516c27de1.png)

Reply:
Fixed
![Screen Shot 2020-08-17 at 12 43 05 PM](https://user-images.githubusercontent.com/55104481/90439254-1aebef80-e08a-11ea-8026-a4a86643a556.png)